src/pages/Signup.js

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Link, useNavigate } from "react-router-dom";
+import { signUp } from "../features/authSlice";
+
+export const Signup = () => {
+  const [form, setForm] = useState({
+    username: "",
+    password: "",
+    confirmPassword: "",
+    firstName: "",
+    lastName: "",
+    email: "",
```

Comment:
Can store this in a temporary variable as emptySignupDetails . And then when the form gets submitted you can reset the signupDetails back to the empty values so that new values can be taken along